design/bridge_design.md

```diff
@@ -0,0 +1,295 @@
+
+> **_NOTE:_** This whole section might change based on the outcome of the Obscuro fees & rewards design. It only outlines a potential solution.
+
+The wormhole implementation requires that a fee is paid for each published message. This is something we should implement exactly as is to prevent people from spamming huge messages on L2 for a little gas cost that results in Obscuro having to pay for them being submitted to the L1!
+
+The problem we will experience is that the native currency on Obscuro L2 will be different from the native currency on L1 - ETH. This means `ETH : OBX` pair volatility might result in losing money on gas costs.
+
+I see a couple of possible solutions to this:
+1. Collect the fees in `WETH` when calling `publishMessage`.
+    * This is a bad user experience. Anytime we want to withdraw we must source `WETH`
+    * It does protect us from losing money when publishing rollups, however.
+2. Using a DEX when it becomes available on Obscuro's L2 we can exchange OBX tokens for the required amount of `WETH`.
+
+An additional insurance fee might be required. It is described in [Security](#MessageBusSecurity)
+
+
+
+### <a name="MessageBusSecurity"></a> Security 
+
+The security of the `MessageBus` is maintained by the `ManagementContract` and the enclave. When the `MessageBus` is secure, then all the downstream apps are secure too. **The maximum achievable security depends on the type of finality the Obscuro L2 has.** 
+
+ * For probabilistic finality - We can be fully secure as L1 block reorgs will reorganize us too.
+ * Fast & hard finality - Block reorgs can lead to instances of the enclave having delivered a message that got thrown away, even when accounting for confirmations.
+
+> **_NOTE:_** The following section is only for fast finality that does not support block reorgs
+
+
+<details>
+  <summary> More on hard & fast finality </summary>
+
+**L2 to L1 Delivery** - The `ManagementContract` needs to ensure that information is exposed through the APIs and/or messages are submitted only after the challenge period of a rollup has passed. This means that anything that shows up there will be final.
+
+**L1 to L2 Delivery** - We need to implement a block confirmation mechanism to secure the `MessageBus` - when the enclave encounters the `LogMessagePublished` it will only create a synthetic transaction for it on the L2 once the required `consistencyLevel` is achieved.
+
+We can also engineer a mechanism to insure delivered messages:
+* When fees are collected for `publishMessage`, part of those fees is funneled to an insurance pool.
+* When a message that has been published gets reorganized the ObscuroDAO can pay for the financial loss from the insurance pool.
+
+> **_NOTE:_** The maximum possible payout should always be less than the penalty for PoS slashing on MainNet. This ensures there is no profit to attacking and claiming insurance.  
```

Comment:
What's missing is an analysis on what happens when someoen sends a message to the l1, which immediately ends up in a message L2->L1, and thus as an instruction. 
How would that work out with reorgs, if this sequence happens exactly before the rollup is emitted.

Do we need to introduce some more delays somewhere?